xcode/Subconscious/Shared/Services/Noosphere/Sphere.swift

```diff
@@ -348,6 +361,25 @@ public final class Sphere: SphereProtocol {
         return changes.toStringArray()
     }
     
+    
+    public func traverse(petname: String) throws -> Sphere? {
+        let identity = try self.getPetname(petname: petname)
+        
+        let sphere = try Noosphere.callWithError(
+            ns_sphere_traverse_by_petname,
+            noosphere.noosphere,
+            sphere,
+            petname
+        )
+        
+        guard let sphere = sphere else {
+            logger.warning("Failed to traverse sphere for \(petname)")
+            return nil
```

Comment:
Should we throw a custom error rather than returning nil? This would make the signature

```swift
func traverse(petname: String) throws -> Sphere
```

I find that it is often more ergonomic to use one or the other style of fallibility, rather than both.

Reply:
Cool, I was wondering about this. I've switched it to throw on a failed traversal instead.